app/share.tsx

```diff
@@ -104,6 +104,9 @@ class Root extends React.Component<{}, IState> {
 			scale,
 			fontScale
 		};
+	}
+
+	UNSAFE_componentWillMount() {
```

Comment:
I'm missing a proper explanation on why this is needed on the description.
Why don't we need this on `/app/index`? https://github.com/RocketChat/Rocket.Chat.ReactNative/blob/f5625cd5f3fa16e2ff06f45728174ecbe6d91fd1/app/index.tsx#L84

Reply:
If your whole state explanation is correct, why does it break it if `currentServer === null` only?
It works properly inside `if (currentServer) {}`.

We used to call `const currentServer = UserPreferences.getString(CURRENT_SERVER);` as asynchronously before #3634 and it might have broken it.
We should try to understand what really happened before making this change you're proposing.
Also, `UNSAFE_componentWillMount` shouldn't be used.